library/common/extensions/filters/http/socket_tag/filter.cc

```diff
@@ -0,0 +1,43 @@
+#include "library/common/extensions/filters/http/socket_tag/filter.h"
+
+#include "envoy/server/filter_config.h"
+
+#include "library/common/network/socket_tag_socket_option_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace SocketTag {
+
+Http::FilterHeadersStatus SocketTagFilter::decodeHeaders(Http::RequestHeaderMap& request_headers,
+                                                         bool) {
+  auto socket_tag_header = Http::LowerCaseString("socket-tag");
+  if (request_headers.get(socket_tag_header).empty()) {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  // The socket-tag header must contain a pair of number separated by a comma, e.g.:
+  // socket-tag: 123,456
+  // The first number contains the UID and the second contains the traffic stats tag.
+  std::string tag_string(request_headers.get(socket_tag_header)[0]->value().getStringView());
```

Comment:
can we reorganize the code to avoid having to call `equest_headers.get(socket_tag_header)` twice?